gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -201,12 +205,18 @@ else if (statement instanceof ExecuteImmediate executeImmediate) {
 
             getNames(statement, tableBuilder, catalogBuilder, schemaBuilder, catalogSchemaBuilder);
             tables = tableBuilder.build();
-            catalogBuilder.addAll(tables.stream().map(q -> q.getParts().getFirst()).iterator());
+
+            Set<QualifiedName> filteredTables = ImmutableSet.copyOf(
+                    tables.stream()
+                            .filter(table -> !temporaryTables.contains(table))
+                            .collect(Collectors.toSet()));
```

Comment:
This can be:
```java
            Set<QualifiedName> filteredTables = tables.stream()
                            .filter(table -> !temporaryTables.contains(table))
                            .collect(toImmutableSet());
```